akka-persistence/src/main/scala/akka/persistence/snapshot/SnapshotStore.scala

```diff
@@ -37,21 +40,49 @@ trait SnapshotStore extends Actor {
       } recover {
         case e ⇒ SaveSnapshotFailure(metadata, e)
       } to (self, p)
-    case evt @ SaveSnapshotSuccess(metadata) ⇒
-      saved(metadata)
-      sender() ! evt // sender is persistentActor
+
+    case evt: SaveSnapshotSuccess ⇒
+      try tryReceivePluginInternal(evt) finally sender() ! evt // sender is persistentActor
     case evt @ SaveSnapshotFailure(metadata, _) ⇒
-      delete(metadata)
-      sender() ! evt // sender is persistentActor
+      try {
+        tryReceivePluginInternal(evt)
+        deleteAsync(metadata)
+      } finally sender() ! evt // sender is persistentActor
+
     case d @ DeleteSnapshot(metadata) ⇒
-      delete(metadata)
+      val p = sender()
+      deleteAsync(metadata) map {
+        case _ ⇒ DeleteSnapshotSuccess(metadata)
+      } recover {
+        case e ⇒ DeleteSnapshotFailure(metadata, e)
+      } to (self, p)
```

Comment:
doesn't matter, up to you (it's technically not needed)

btw I normally use `pipeTo` instead of `to`, is there a difference?